packages/css/modal.css

```diff
@@ -28,90 +30,65 @@
       fade-in 300ms ease-in-out;
   }
 
-  & > hr {
-    margin: var(--ds-spacing-3) 0 !important;
-  }
-
   @media (max-width: 40rem) {
-    & {
-      min-width: 100%;
-      max-width: 100%;
-      border-radius: 0;
-    }
+    min-width: 100%;
+    max-width: 100%;
+    border-radius: 0;
   }
 
   @media (prefers-reduced-motion: reduce) {
-    &[open] {
-      animation: none;
-    }
-
+    &[open],
     &::backdrop {
       animation: none;
     }
   }
 
-  &.ds-modal--lock-scroll {
-    overflow: hidden;
-  }
-
-  .ds-modal__header {
-    display: flex;
-    justify-content: space-between;
-    flex-direction: column;
-    padding: var(--dsc-modal-header-padding);
-    gap: var(--ds-spacing-1);
-
-    &:not(:has(> .ds-modal__header__button)) {
-      --dsc-modal-header-padding: var(--ds-spacing-6) var(--ds-spacing-6) var(--ds-spacing-2) var(--ds-spacing-6);
-    }
-
-    .ds-modal__header__button {
-      position: absolute;
-      top: var(--ds-spacing-3);
-      right: var(--ds-spacing-3);
-      color: var(--dsc-modal-color);
-
-      &::before {
-        content: '';
-        background: currentcolor;
-        height: 1.5em;
-        width: 1.5em;
-        mask: center/contain no-repeat
-          url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' width='1em' height='1em' fill='none' viewBox='0 0 24 24' focusable='false' role='img'%3E%3Cpath fill='currentColor' d='M6.53 5.47a.75.75 0 0 0-1.06 1.06L10.94 12l-5.47 5.47a.75.75 0 1 0 1.06 1.06L12 13.06l5.47 5.47a.75.75 0 1 0 1.06-1.06L13.06 12l5.47-5.47a.75.75 0 0 0-1.06-1.06L12 10.94z'%3E%3C/path%3E%3C/svg%3E");
-      }
+  /* Close button */
+  & > form[method='dialog']:first-child > button:only-child {
+    --margin-top-right: calc(var(--dsc-modal-padding-invert) + var(--dsc-modal-close-margin));
+
+    float: right;
+    margin: var(--margin-top-right) var(--margin-top-right) var(--dsc-modal-close-margin) var(--dsc-modal-close-margin);
+    color: inherit;
+
+    &::before {
+      content: '';
+      background: currentcolor;
+      height: var(--ds-spacing-6);
+      width: var(--ds-spacing-6);
+      mask: center/contain no-repeat
+        url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' width='1em' height='1em' fill='none' viewBox='0 0 24 24' focusable='false' role='img'%3E%3Cpath fill='currentColor' d='M6.53 5.47a.75.75 0 0 0-1.06 1.06L10.94 12l-5.47 5.47a.75.75 0 1 0 1.06 1.06L12 13.06l5.47 5.47a.75.75 0 1 0 1.06-1.06L13.06 12l5.47-5.47a.75.75 0 0 0-1.06-1.06L12 10.94z'%3E%3C/path%3E%3C/svg%3E");
     }
   }
+}
 
-  .ds-modal__footer {
-    display: flex;
-    align-items: center;
-    gap: var(--ds-spacing-4);
-    padding: var(--dsc-modal-footer-padding);
-  }
+.ds-modal__header {
+  border-bottom: var(--dsc-modal-divider);
```

Comment:
Did you clear always adding a border with design?

Reply:
Yes, this is verified with design ☺️